x-pack/plugins/security_solution/server/endpoint/routes/metadata/handlers.ts

```diff
@@ -167,52 +181,76 @@ export const getMetadataRequestHandler = function (
 export async function getHostData(
   metadataRequestContext: MetadataRequestContext,
   id: string,
-  queryStrategyVersion?: MetadataQueryStrategyVersions
+  queryStrategyVersion?: MetadataQueryStrategyVersions,
+  includeAgentData?: boolean
 ): Promise<HostInfo | undefined> {
+  if (!metadataRequestContext.savedObjectsClient) {
+    return undefined;
+  }
   const queryStrategy = await metadataRequestContext.endpointAppContextService
     ?.getMetadataService()
-    ?.queryStrategy(
-      metadataRequestContext.requestHandlerContext.core.savedObjects.client,
-      queryStrategyVersion
-    );
-
+    ?.queryStrategy(metadataRequestContext.savedObjectsClient, queryStrategyVersion);
   const query = getESQueryHostMetadataByID(id, queryStrategy!);
-  const hostResult = queryStrategy!.queryResponseToHostResult(
-    await metadataRequestContext.requestHandlerContext.core.elasticsearch.legacy.client.callAsCurrentUser(
-      'search',
-      query
-    )
-  );
+
+  if (
+    !metadataRequestContext.esClient &&
+    !metadataRequestContext.requestHandlerContext?.core.elasticsearch.client
+  ) {
+    return undefined;
+  }
+
+  const esClient = (metadataRequestContext?.esClient ??
+    metadataRequestContext.requestHandlerContext?.core.elasticsearch
+      .client) as IScopedClusterClient;
+  const response = await esClient.asCurrentUser.search<HostMetadata>(query);
+
+  const hostResult = queryStrategy!.queryResponseToHostResult(response.body);
+
   const hostMetadata = hostResult.result;
   if (!hostMetadata) {
     return undefined;
   }
 
-  const agent = await findAgent(metadataRequestContext, hostMetadata);
+  /* Skip agent data if we only need meta data, by doing this
+   **  we can avoid users get 404 error from findAgent and
+   ** therefore unable to get endpoint meta data
+   */
```

Comment:
the 404 bit might only be relevant to your team's calling context. This comment doesn't make much sense from the endpoint calling perspective.

Can probably just note that `Enriching with agent data is optional for some callers`